src/analyses/variable-sensitivity/abstract_object.h

```diff
@@ -288,5 +302,34 @@ bool abstract_objectt::merge_maps(
   return modified;
 }
 
+template<typename keyt>
+bool abstract_objectt::merge_shared_maps(
```

Comment:
Currently when this function is invoked, a clone of `m1` is passed as `out_map`. The current version would only work though if `out_map` were the empty map (but then we would not have sharing). We also need to remove those items that are only in `m1`/`out_map`. We can do this by using `get_delta_view(..., false)`. Then we should collect the keys of the elements to erase in a `keyst` vector, which after the loop we can pass to `erase_all()` (operations that add or erase keys from the map invalidate a `view`/`delta_view`.

The same issue also appears in `merge_maps()`.